### DISCUSSION TITLE
[General] Blocked by cloudflare

### DISCUSSION THREAD

**Discussion by ZONTeek** [zkSync-Community-Hub/zksync-developers#84] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

I'm developing wallet application and trying to load completed transactions on scroll.

API: https://zksync2-mainnet-explorer.zksync.io/transactions

### What is the specific issue or error you're encountering?

Somehow I got blocked by Cloudflare. 
My guess is onScroll event triggered multiple times and I sent too much requests leaded to block

### Can you share the error messages or logs you're receiving, if any?

<img width="676" alt="image" src="https://github.com/zkSync-Community-Hub/zkync-developers/assets/57589941/2abc3d67-9f1c-46d8-af82-5da1819a3ba3">

### Have you made any recent changes to the contract before encountering this issue?

No

### Are there any external libraries or contracts that your contract interacts with?

https://www.npmjs.com/package/zksync
https://www.npmjs.com/package/zksync-web3

### Can you provide the relevant portions of your contract code where the issue is occurring?

No

### Have you tried to isolate the problem, and if so, what were the results?

mainnet.era , rpc.ankr and api.trongrid.io still working. 

### What steps have you already taken to try to resolve the issue?

Decrease RPS

**Comment by MexicanAce** [comment, marked as answer]
Hi @ZONTeek 👋 

It looks like your use of the Block Explorer APIs was tagged as a potential attack (possibly due to the high request rate from your IP address). These measures are in place to ensure that these Production systems are less susceptible to Denial of Service attacks.

Were you able to successfully reduce the number of requests per second from your application to https://zksync2-mainnet-explorer.zksync.io/transactions ? Are you still experiencing this issue?

**Comment by bricnounlerumark** [comment]
It seems like your application might be triggering too many requests to the API endpoint, leading to being blocked by Cloudflare. To resolve this issue, you can try implementing rate limiting on your onScroll event to prevent excessive API requests. Additionally, you may want to optimize your code to fetch transactions more efficiently, such as implementing pagination or caching.

**Comment by whaliop** [comment]
Issue: Blocked by Cloudflare when loading completed transactions on scroll from the zkSync mainnet explorer API.

Possible cause: Multiple onScroll events triggering too many requests, leading to a Cloudflare block.

Steps taken: Decreased request per second (RPS) to mitigate the issue.

Solution: Implement rate limiting or debounce the onScroll event to prevent excessive requests.

**Comment by eleykelly** [comment]
Issue: Cloudflare blocking requests due to high request rate triggered by onScroll event loading completed transactions.

Potential Solutions:
1. Implement rate limiting or debounce mechanism on the onScroll event to reduce request frequency.
2. Check Cloudflare logs for more detailed information on the block reason and adjust request rate accordingly.
3. Consider caching or optimizing API requests to minimize the number of requests sent.

**Comment by latoyatidwell** [comment]
You may have triggered Cloudflare's protection due to excessive requests. Consider optimizing your code to reduce the frequency of API calls or implement caching mechanisms to avoid hitting rate limits.

**Comment by ngopriyanka** [comment]
It appears that the user has been blocked by Cloudflare while attempting to load completed transactions on scroll using the ZkSync API. This may be due to triggering the onScroll event multiple times and sending too many requests, leading to the block. The user has tried decreasing the RPS to resolve the issue. Other endpoints such as mainnet.era, rpc.ankr, and api.trongrid.io are still working.
